### PR TITLE
fix: downgrade opentelemetry-operations-go to 0.34.1

### PR DESCRIPTION
**Pull request by liufuyang**
as we see memory leak type of issue and a constant increase of bytes to google with updated version of opentelemetry-operations-go.

This reverts some of the changes made here https://github.com/einride/cloudrunner-go/pull/382

The issue seen from a backend service could be like:

vehicle-service
image

own-service
image

vehicle-service CPU profile
image

https://console.cloud.google.com/profiler;timespan=1h;end=2023-02-14T11:30:00.000Z/vehicle-eu/cpu?project=e-vehicle-service-prod&pli=1

Also could see the backend service's metric will break when updated...
image

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -40,8 +40,8 @@ require (
 	cloud.google.com/go/compute v1.18.0 // indirect
 	cloud.google.com/go/monitoring v1.12.0 // indirect
 	cloud.google.com/go/trace v1.8.0 // indirect
-	github.com/GoogleCloudPlatform/opentelemetry-operations-go/detectors/gcp v1.11.1 // indirect
-	github.com/GoogleCloudPlatform/opentelemetry-operations-go/internal/resourcemapping v0.35.1 // indirect
+	github.com/GoogleCloudPlatform/opentelemetry-operations-go/detectors/gcp v1.11.0 // indirect
```
Comment (liufuyang) [inline go.mod:43]: Tried to put this to `1.10.1` but go mod tidy updates it.

**Comment by liufuyang** [comment]
Some other packages needed to downgrade as well to fix build error:

```
Error: ../../../go/pkg/mod/github.com/!google!cloud!platform/opentelemetry-operations-go/exporter/metric@v0.34.1/cloudmonitoring.go:48:9: cannot use newMetricExporter(&o) (value of type *metricExporter) as type "go.opentelemetry.io/otel/sdk/metric".Exporter in return statement:
*metricExporter does not implement "go.opentelemetry.io/otel/sdk/metric".Exporter (missing Aggregation method)
```

**Comment by thall** [comment]
Cant we just revert https://github.com/einride/cloudrunner-go/pull/382 ?

**Comment by liufuyang** [comment]
@thall This is more or less a manual revert of #382 but only touch on the places that need to be reverted.

We can perhaps revert #382 as well but I guess it also needs to revert https://github.com/einride/cloudrunner-go/pull/394 at the same time?

**Comment by thall** [comment]
I think its easier just revert the whole PR rather than trying to cherry-pick stuff from it.
Oh you have already merged one PR, either you revert that and then the original, or you continue the path you are on.

**Comment by liufuyang** [comment]
Tried to revert one by one from the end but got some conflicts... Will not touch it for now, and feel free to make another PR on your side to fix it if you find a good/easy way to revert those commits back :)

**Review by ericwenn — APPROVED** [review]

**Comment by liufuyang** [comment]
We can try update again when this https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/591 is merged

**Pull request merged**